crates/optimism/rpc/src/eth/mod.rs

```diff
@@ -129,20 +178,48 @@ impl<N: RpcNodeCore, Rpc: RpcConvert> OpEthApi<N, Rpc> {
         };
 
         let Some(rx) = self.inner.pending_block_rx.as_ref() else { return Ok(None) };
-        let pending_block = rx.borrow();
-        let Some(pending_block) = pending_block.as_ref() else { return Ok(None) };
 
-        let now = Instant::now();
+        let (expires_at, last_flashblock_index, _parent_hash, pending) = {
+            let pending_block = rx.borrow();
+            let Some(pending_block) = pending_block.as_ref() else { return Ok(None) };
+
+            let now = Instant::now();
+
+            // Is the pending block not expired and latest is its parent?
+            if pending.evm_env.block_env.number != U256::from(pending_block.block().number()) ||
+                parent.hash() != pending_block.block().parent_hash() ||
+                now > pending_block.expires_at
+            {
+                return Ok(None);
+            }
```

Comment:
this check is now bypassed in the next step

what we can do is extract obtaining the block to `async fn flashbock()`, then we don't really need to modify this function much